### ISSUE TITLE
Allow newlines for `function-calc-no-unspaced-operator` rule

### ISSUE THREAD

**Issue by keithamus** [stylelint/stylelint#857]
The `function-calc-no-unspaced-operator`, when configured to warn, will warn when using an operator without spaces, e.g.
```css
margin-top: calc(var(--some-variable)+var(--some-other-variable));
```
The config to generate a warning would be:
```json
{
  "rules": {
    "function-calc-no-unspaced-operator": true,
  }
}
```

However, it also fails fit he whitespace used is a newline:
```css
margin-top: calc(var(--some-variable) +
  var(--some-other-variable));
```

It would be nice for this rule to also allow newlines as a valid form of whitepsace (perhaps optionally?).

### Other info:
Stylelint: 4.4.0
Cli: `stylelint src/*.css"

**Comment by davidtheclark** [comment]
I agree. This seems to me like a good feature request.

**Comment by jeddy3** [comment]
Sgtm.

> It would be nice for this rule to also allow newlines as a valid form of whitepsace (perhaps optionally?).

I think we can change the default behaviour of the rule to accept newlines (rather than introduce an option), as it's more permissive than we have right now.  Marking as "help-wanted" :)

**Comment by keithamus** [comment]
Happy to put the work in including tests and documentation, but I'm unfamiliar with the parser API - I'm guessing I'll have to add something around [here](https://github.com/stylelint/stylelint/blob/a2fab85a30ee264c09e96cb916b64353e24ac8bc/src/rules/function-calc-no-unspaced-operator/index.js#L59) - or would I need to edit the properties of whitespaceChecker? Guidance appreciated.

**Comment by davidtheclark** [comment]
Tests and docs would be absolutely fantastic. I'd have to spend some minutes looking into the rule to see what might need to change -- probably will get to it this weekend.

**Comment by keithamus** [comment]
@davidtheclark if you can point me in the general direction, I could make a start at it, but right now I'm not even sure where to start :blush:

**Comment by davidtheclark** [comment]
@keithamus: In terms of tests & docs I'd say it's pretty clear (or did you have questions about that?). For changing the rule, I think the thing to do would be to rewrite the logic of this rule to *not* use `whitespaceChecker` — which only provides minimal value for this particular rule and would make it awkward to check for arbitrary whitespace. Does that make sense, give you a place to start with?

**Comment by jeddy3** [comment]
Would [`function-whitespace-after`](https://github.com/stylelint/stylelint/blob/master/src/rules/function-whitespace-after/index.js) be a good whitespace rule to refer to here?

**Comment by davidtheclark** [comment]
I actually wonder if we should create a new rule, `function-calc-operator-around`, which has two options: `"single-space"` and `"any-space"`. If you want to enforce one single space, you have an option; and if you want more flexibility like @keithamus you can also do that. Thoughts?

**Comment by keithamus** [comment]
I'm happy to do whatever you all think is best. Could an option be added to the existing rule perhaps?

**Comment by davidtheclark** [comment]
Yeah, I guess an option would be fine, too. `any-whitespace` maybe?

The reason for my second-guessing is that I was actually looking into implementing this last night and as I was modifying the tests realized that there people you probably *want* to enforce only a single space (no tabs or newlines) around the operators. So we should try to accommodate both.

**Comment by jeddy3** [comment]
> So we should try to accommodate both.

Agreed.

`function-calc-operator-around` is more consistent, but it would require a deprecation. I'm happy to go with an option if you are?

Do we make that option boolean e.g. `allowNewline: true`. Default behaviour would still check for just a single space. This option would allow a single space or a newline.
